.travis.yml

```
@@ -224,17 +224,10 @@ jobs:
   - <<: *stage_deploy
     if: tag IS present
     <<: *python_3_7_mixture
-    install: skip
-    script: skip
-    deploy:
-      provider: pypi
-      on:
-        all_branches: true
-      user: jaraco
-      password:
-        secure: RAfz06AINvz7bfij/YhfkAreRqamgxS8a6jSRNxntYhtJke3ZszUbIDag8+n1I+G5XT2LnMhHqPNR7Plc+AeMz7VXTuy+b81Li5kse20NYlPhd7mBVmTUpXtqYQashV5J39F4qkATBLznCOrMEomM07VTXjO/o2hmQuXniab2Uo=
-      distributions: dists
-      skip_cleanup: true
+    install:
```

This is not needed, just add `tox-venv` to `tox.ini` and set
```suggestion
    env:
      TOXENV: release
```

---

I'm not sure I want to invoke `tox -m tox --notest` or `*python_ssl_openssl_version` or `python -m OpenSSL.debug` during `install`, which is why I chose to override it.

Also, I'm not sure about adding `tox-venv` to `tox.ini`. There may be bootstrapping issues. If `tox` needs a later version or needs to install `tox-venv`, then it will create a `.tox/.tox` env in which to install those packages. If `tox-venv` isn't already installed (or is), that may affect how `.tox/.tox` is created. The only technique I trust is to install `tox-venv` alongside `tox` prior to invoking `.tox`. I'm open to experimenting with other techniques, but I don't want to adopt those risks as part of this change.

---

Okay, so let's change this to installing `tox-venv` and invoking `tox` with `--notest` because `--notest` allows us to clearly separate envs bootstrap (hence the install step) from actually running main commands (script).
Using `env` allows us to unify interactions with tox across the config file.

---

Oh, I see. By running `--notest`, you get bootstrapping of the environment(s) before running their commands. That's interesting and does make sense. I'm reluctant to add complication, but I can see the benefit there. Let's explore that after validating this simpler approach.

---

Yep, that was my point :)

---

Oh, this discussion got distributed over comments... https://github.com/cherrypy/cheroot/pull/226#discussion_r328815933